sdk/cosmosdb/cosmos/src/utils/nonStreamingOrderByPriorityQueue.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import PriorityQueue from "priorityqueuejs";
+
+export class NonStreamingOrderByPriorityQueue<T> {
```

Comment:
The name of the class might be misleading. Since the size-based evacuation is not part of contract of Priority Queue. Also since this implementation is independent of the Streaming query. A generic name i.e `FixedSizedPriorityQueue` might be more appropriate.

Reply:
updated